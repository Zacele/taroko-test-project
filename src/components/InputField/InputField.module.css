.container {
  position: relative;
  margin-top: 14px;
}

.input {
  width: 100%;
  border: none;
  border-bottom: 1px solid #ccc;
  outline: none;
  padding: 10px 0;
  background: transparent;
  font-size: 16px;
  color: #333;
  transition: border-color 0.2s;
}

.input:focus {
  border-bottom: 2px solid #6200ea;
}

.input:focus + .label,
.input:not(:placeholder-shown) + .label {
  transform: translateY(-20px);
  font-size: 12px;
  color: #6200ea;
}

.label {
  position: absolute;
  top: 0;
  left: 0;
  color: #999;
  font-size: 16px;
  transition: all 0.2s ease-out;
  pointer-events: none;
}
