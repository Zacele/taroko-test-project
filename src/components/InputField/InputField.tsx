'use client';
import React from 'react';
import styles from './InputField.module.css';

interface InputFieldProps {
  label: string;
  value: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  placeholder?: string;
  type?: string;
}

const InputField: React.FC<InputFieldProps> = ({
  label,
  value,
  onChange,
  placeholder,
  type = 'text',
}) => {
  return (
    <div className={styles.container}>
      <input
        className={styles.input}
        value={value}
        onChange={onChange}
        placeholder={placeholder}
        type={type}
        id={`input-${label}`}
      />
      <label htmlFor={`input-${label}`} className={styles.label}>
        {label}
      </label>
    </div>
  );
};

export default InputField;
