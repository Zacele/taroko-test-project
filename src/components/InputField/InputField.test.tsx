import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import InputField from './InputField';

describe('InputField', () => {
  const mockOnChange = jest.fn();

  beforeEach(() => {
    mockOnChange.mockClear();
  });

  test('renders correctly with all props', () => {
    const { asFragment } = render(
      <InputField
        label="Test Label"
        value="Test Value"
        onChange={mockOnChange}
        placeholder="Test Placeholder"
        type="text"
      />,
    );

    expect(screen.getByLabelText(/test label/i)).toBeInTheDocument();
    expect(screen.getByRole('textbox')).toHaveValue('Test Value');
    expect(
      screen.getByPlaceholderText(/test placeholder/i),
    ).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('calls onChange when input changes', () => {
    render(<InputField label="Test Label" value="" onChange={mockOnChange} />);
    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: 'New Value' } });

    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
