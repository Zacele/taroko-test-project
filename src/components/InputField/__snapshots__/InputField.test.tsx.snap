// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders correctly with all props 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <input
      class="input"
      id="input-Test Label"
      placeholder="Test Placeholder"
      type="text"
      value="Test Value"
    />
    <label
      class="label"
      for="input-Test Label"
    >
      Test Label
    </label>
  </div>
</DocumentFragment>
`;
